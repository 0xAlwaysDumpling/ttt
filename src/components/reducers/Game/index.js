
import { loadState } from '../../localstorage';
import { initialState, resetGameState } from './constants';
import { postMove } from '../../api';

/* 
  To mock API call I needed to do actions MOVE_REQUEST, MOVE_SUCCESS, MOVE_FAILED 
  In the interest of scope and time I just made a function call postMove

*/
const GameState = (state = initialState, action) => {
  switch (action.type) {
    case 'LOAD':
      const appState = loadState();
      return appState ? appState : initialState;
    case 'MOVE':
      if (state.gameHistory.hasEnded) return state;
      else return postMove(state, action)
    case 'RESET':
      return {
        ...state,
        ...resetGameState
      };
    default:
      return state;
  }
}

export default GameState;