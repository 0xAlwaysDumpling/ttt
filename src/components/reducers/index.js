import { combineReducers } from 'redux';
import GameState from './Game';


export default combineReducers({
  GameState
});

