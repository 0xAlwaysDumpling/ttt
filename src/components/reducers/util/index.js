import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react';
import rootReducer from '../index';
import { initialState as GameInitialState } from '../Game/constants'

const initialState = {
  GameState: GameInitialState
}

export const renderWithRedux = (
  component,
) => {
  const store = createStore(rootReducer, initialState)
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  }
}
