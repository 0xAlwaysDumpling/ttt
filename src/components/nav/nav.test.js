import React from 'react'
import { cleanup } from '@testing-library/react'
import Nav from './index';
import { renderWithRedux } from '../reducers/util';

afterEach(cleanup)

it('should render with out crashing', () => {
  renderWithRedux(<Nav />)
});

it('should take a snapshot of Nav', () => {
  const { asFragment } = renderWithRedux(<Nav />)

  expect(asFragment(<Nav />)).toMatchSnapshot()
});
