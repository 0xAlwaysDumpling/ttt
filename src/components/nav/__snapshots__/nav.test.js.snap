// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of Nav 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm eaZpM"
  >
    <p
      class="sc-AxirZ hdKPDr"
    >
      P1: 0 W - 0 L
    </p>
    <p
      class="sc-AxirZ hdKPDr"
    >
      P2: 0 W - 0 L
    </p>
    <p
      class="sc-AxirZ hdKPDr"
    >
      Draws: 0
    </p>
  </div>
</DocumentFragment>
`;
